Task1.1.TrainingWebsite/Task1.1.TrainingWebsite/Entities/TrainingLesson.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Task1._1.TrainingWebsite.Entities.TrainingMaterial;
+using Task1._1.TrainingWebsite.Enums;
+using Task1._1.TrainingWebsite.Extensions;
+using Task1._1.TrainingWebsite.Interfaces;
+
+namespace Task1._1.TrainingWebsite.Entities
+{
+    internal class TrainingLesson : EntityBase, IVersionable
+    {
+        public List<EntityBase> TrainingMaterials = new List<EntityBase>();
+        private ulong _version;
+        public byte[] ReadVersion(ulong version)
+        {
+            return BitConverter.GetBytes(_version);
+        }
+        public ulong SetVersion(ulong version)
+        {
+           return _version = version;
+        }
+        public TrainingLesson(string description, List<EntityBase> trainingMaterials) : base(description)
```

Comment:
```suggestion
        public TrainingLesson(string description, IEnumerable<EntityBase> trainingMaterials) : base(description)
```
IEnumerable allows to pass any collection without type casting. It is a common practice to have most generic types on input and most concrete on output.

Reply:
I will then need to refactor the TrainingLesson class so that I could iterate it with foreach (var item in trainingLesson) .... ?

Reply:
You can keep field of trainingMaterials as any type of collection, just use enumerable for an input. 

```
 public TrainingLesson(string description, IEnumerable<EntityBase> trainingMaterials) : base(description)
{
   this.trainingMaterials = trainingMaterials.ToList();
}
```